pkg/agent/flags/flag_controller.go

```diff
@@ -79,6 +80,7 @@ func (fc *FlagController) Update(kvPairs map[string]string) ([]string, error) {
 
 	// Changed keys is the union of updated keys and deleted keys
 	changedKeys := append(updatedKeys, deletedKeys...)
+	sort.Strings(changedKeys)
```

Comment:
Do we need this to be deterministic here, or just in tests? I'm not sure how I feel about determinism.